music21/romanText/tsvConverter.py

```diff
@@ -0,0 +1,670 @@
+
+    for key in characterDict:  # Both major and minor
+        preString = preString.replace(key, characterDict[key])
+
+    if not minor:  # == False:
+        return preString
+    else:  # minor == True:
+        if direction == 'm21-DCML':
+            search = 'b'
+            insert = '#'
+        elif direction == 'DCML-m21':
+            search = '#'
+            insert = 'b'
+        else:  # Should be redundant: covered above.
+            raise ValueError("Direction must be 'm21-DCML' or 'DCML-m21'.")
+
+        if 'vii' in preString.lower():
+            position = preString.lower().index('vii')
+            prevChar = preString[position-1] # the previous character,  # / b.
+            if prevChar == search:
+                postString = preString[:position-1]+preString[position:]
+            else:
+                postString = preString[:position]+insert+preString[position:]
+        else:
+            postString = preString
+
+    return postString
+
+def getLocalKey(local_key, global_key, convert=False):
+    '''
+    Re-casts comparative local key (e.g. 'V of G major') in its own terms ('D').
+    >>> romanText.tsvConverter.getLocalKey('V', 'G')
+    'D'
+    >>> romanText.tsvConverter.getLocalKey('ii', 'C')
+    'd'
+
+    By default, assumes an m21 input, and operates as such:
+    >>> romanText.tsvConverter.getLocalKey('vii', 'a')
+    'g#'
+
+    Set convert=True to convert from DCML to m21 formats. Hence;
+    >>> romanText.tsvConverter.getLocalKey('vii', 'a', convert=True)
+    'g'
+    '''
+
+    if convert==True:
+        local_key = characterSwaps(local_key, minor=is_minor(global_key[0]), direction='DCML-m21')
+
+    asRoman = roman.RomanNumeral(local_key, global_key)
+    rt = asRoman.root().name
+    if asRoman.isMajorTriad():
+        newKey = rt.upper()
+    elif asRoman.isMinorTriad():
+        newKey = rt.lower()
+
+    return newKey
+
+def vLocalKey(rn, local_key):
+    '''
+    Separates comparative roman numeral for tonicisiations like 'V/vi' into the component parts of
+    - a roman numeral (V) and
+    - a (very) local key (vi)
+    and expresses that very local key in relation to the local key also called (DCML column 11).
+
+    >>> romanText.tsvConverter.getLocalKey('vi', 'C')
+    'a'
+    >>> romanText.tsvConverter.vLocalKey('V/vi', 'C')
+    'a'
+    '''
+
+    if '/' not in rn:
+        very_local_as_key = local_key
+    else:
+        position = rn.index('/')
+        very_local_as_roman = rn[position+1:]
+        very_local_as_key = getLocalKey(very_local_as_roman, local_key)
+
+    return very_local_as_key
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+
+    def runTest(self):
+        pass
+
+    def testTsvHandler(self):
+
+        name = 'tsvEgOp18No1Mov3.tsv'
+        path = common.getSourceFilePath() / 'romanText' / name
+
+        importedTsv = TsvHandler(path)
```

Comment:
importedTsv seems to imply it is a data structure rather than an object.  `handler` would be a better name.